packages/client/signers/src/communications/schemas.ts

```diff
@@ -107,9 +110,15 @@ export const GetPublicKeyPayloadSchema = {
 export const GetStatusPayloadSchema = {
     request: AuthenticatedEventRequest,
     response: ResultResponse(
-        z.object({
-            signerStatus: z.enum(["ready", "new-device"]).describe("Current status of the signer"),
-        })
+        z.union([
+            z.object({
+                signerStatus: z.enum(["ready"]).describe("Current status of the signer"),
+                publicKeys: PublicKeyMappingSchema.describe("The public keys of the created signer"),
+            }),
+            z.object({
+                signerStatus: z.enum(["new-device"]).describe("Current status of the signer"),
+            }),
+        ])
```

Comment:
Should we have a single source of truth for these schema since they're shared across multiple responses?